kernel/kernel-api/src/test/scala/io/delta/kernel/internal/TableConfigSuite.scala

```diff
@@ -0,0 +1,131 @@
+    new Metadata(
+      java.util.UUID.randomUUID().toString,
+      Optional.empty(),
+      Optional.empty(),
+      new Format(),
+      "",
+      null,
+      stringArrayValue(Collections.emptyList()),
+      Optional.empty(),
+      VectorUtils.stringStringMapValue(Collections.emptyMap())
+    )
+  }
+
+  test("Coordinated Commit Related Properties from Metadata") {
+    val expMap = Map("key1" -> "string_value", "key2Int" -> "2", "key3ComplexStr" -> "\"hello\"")
+    val engine1 = mockEngine(jsonHandler = new KeyValueJsonHandler(expMap))
+    val m1 = getEmptyMetadata.withNewConfiguration(
+      Map(
+        TableConfig.COORDINATED_COMMITS_COORDINATOR_NAME.getKey -> "cc-name",
+        TableConfig.COORDINATED_COMMITS_COORDINATOR_CONF.getKey ->
+          """{"key1": "string_value", "key2Int": "2", "key3ComplexStr": "\"hello\""}""",
+        TableConfig.COORDINATED_COMMITS_TABLE_CONF.getKey ->
+          """{"key1": "string_value", "key2Int": "2", "key3ComplexStr": "\"hello\""}""").asJava
+    )
+    assert(TableConfig.COORDINATED_COMMITS_COORDINATOR_NAME.fromMetadata(engine1, m1) ===
+      Optional.of("cc-name"))
+
+    assert(TableConfig.COORDINATED_COMMITS_COORDINATOR_CONF.fromMetadata(engine1, m1) ===
+      expMap.asJava)
+
+    assert(TableConfig.COORDINATED_COMMITS_TABLE_CONF.fromMetadata(engine1, m1) ===
+      expMap.asJava)
+
+    val engine2 = mockEngine(jsonHandler = new RuntimeExceptionJsonHandler)
+    val m2 = getEmptyMetadata.withNewConfiguration(
+      Map(
+        TableConfig.COORDINATED_COMMITS_COORDINATOR_NAME.getKey -> null,
+        TableConfig.COORDINATED_COMMITS_COORDINATOR_CONF.getKey ->
+          """{"key1": "string_value", "key2Int": "2""",
+        TableConfig.COORDINATED_COMMITS_TABLE_CONF.getKey ->
+          """{"key1": "string_value", "key2Int": "2""").asJava
+    )
+    assert(TableConfig.COORDINATED_COMMITS_COORDINATOR_NAME.fromMetadata(engine2, m2) ===
+      Optional.empty())
+    intercept[RuntimeException] {
+      TableConfig.COORDINATED_COMMITS_COORDINATOR_CONF.fromMetadata(engine2, m2)
+    }
+    intercept[RuntimeException] {
+      TableConfig.COORDINATED_COMMITS_TABLE_CONF.fromMetadata(engine2, m2)
+    }
+  }
+}
+
+class KeyValueJsonHandler(map: Map[String, String]) extends BaseMockJsonHandler {
+  val schema: StructType =
+    new StructType().add("config", new MapType(StringType.STRING, StringType.STRING, true))
+  override def parseJson(
+    jsonStringVector: ColumnVector,
+    outputSchema: StructType,
+    selectionVector: Optional[ColumnVector]): ColumnarBatch = {
+    new ColumnarBatch {
+      override def getSchema: StructType = schema
+
+      override def getColumnVector(ordinal: Int): ColumnVector = {
+        ordinal match {
+          case 0 => mapTypeVector(map.keys.toSeq, map.values.toSeq) // path
+        }
+      }
+
+      override def getSize: Int = 1
+    }
+  }
+
+  def mapTypeVector(keys: Seq[String], values: Seq[String]): ColumnVector = {
```

Comment:
Can we use `VectorUtils.stringStringMapValue`?

Reply:
Yes! It can simplify the code.